website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,388 @@
+The hard part is: we can't just upload the bytes of a .png or .jpg file! These formats have a lot of unnecessary data, like headers and most importantly
+compression, which the GPU generally cannot understand.
+
+To load an image file as a byte array, we will first need a external library. In our case, we will use StbImageSharp.
+
+First of all, install the library using NuGet. To do so, run the following command:
+```
+dotnet add package StbImageSharp
+```
+
+After installing, add the following reference on the top of your code file:
+```c#
+using StbImageSharp;
+```
+
+At the top of your class, declare a `uint` to store the ID of the OpenGL texture object:
+```c#
+private static uint _texture;
+```
+
+Returning to your `OnLoad` method, add this line at the end of the method to create, and bind the texture.
+That `ActiveTexture` call is telling OpenGL that we are wanting to use the first texture slot.
+```c#
+_texture = _gl.GenTexture();
+_gl.ActiveTexture(TextureUnit.Texture0);
+_gl.BindTexture(TextureTarget.Texture2D, _texture);
+```
+
+After that, we need to load the image. You can do it with the following line:
+```c#
+// ImageResult.FromMemory reads the bytes of the .png file and returns all its information!
+ImageResult result = ImageResult.FromMemory(File.ReadAllBytes("silk.png"), ColorComponents.RedGreenBlueAlpha);
+```
+
+Now, with the image in memory, we need to upload the data to the GPU texture. To dot hat, we need a pointer for our bytes, the width, and the height of
+the texture. None of it is hardcoded, you get all this information inside `ImageResult`.
+```c#
+// Define a pointer to the image data
+fixed (byte* ptr = result.Data)
+    // Here we use "result.Width" and "result.Height" to tell OpenGL about how big our texture is.
+    _gl.TexImage2D(TextureTarget.Texture2D, 0, InternalFormat.Rgba, (uint)result.Width,
+        (uint)result.Height, 0, PixelFormat.Rgba, PixelType.UnsignedByte, ptr);
+```
+
+You don't need to understand what all these values mean! The OpenGL documentation explains it all in depth. Right now
+let's just continue with the basics!
+
+Now, with the texture data sent off to the GPU, we need to configure the texture parameters. This is one of the most important steps. It is with these
+parameters that the shader knows how to sample colors from the texture using our provided texture coordinates.
+
+For now, you don't need to understand what these do. We will elaborate on it in the future. Just add these lines after you upload the texture data.
+```c#
+_gl.TextureParameter(_texture, TextureParameterName.TextureWrapS, (int) TextureWrapMode.Repeat);
+_gl.TextureParameter(_texture, TextureParameterName.TextureWrapT, (int) TextureWrapMode.Repeat);
+_gl.TextureParameter(_texture, TextureParameterName.TextureMinFilter, (int) TextureMinFilter.Linear);
+_gl.TextureParameter(_texture, TextureParameterName.TextureMagFilter, (int) TextureMagFilter.Linear);
```

Comment:
Consider using `TexParameter` instead of `TextureParameter`, since I don't think we're not using named functions anywhere else (correct me if I'm wrong)

Reply:
`TextureParameter` isn't, anyway, easier to use and read?

Reply:
Yes, but `glTextureParameter` is only available on OpenGL 4.5, whereas these tutorials seem to be targeting OpenGL 3.3

Reply:
Yeahm the tutorials (the ones I've written anyway) have all targeted GL 3.3.

Reply:
tbh i don't know how to use `TexParameter`, i usually use `TextureParameter` and the source uses `TextureParameter` also, someone can help me with it?

Reply:
Whoops, seems like that was a mistake on my part. Useage of `TexParameter` is the same as `TextureParameter`, it just acts on the currently bound texture, so you don't pass it the texture handle.

Reply:
in the reality i need to make some conversions and use different enuns but it's fine